### DISCUSSION TITLE
How do I access expenseRatio for ETFs

### DISCUSSION THREAD

**Discussion by ntigern** [ranaroussi/yfinance#2012] [Q&A]
Hi all,
I am hoping to add expense ratio to the data that I get from yfinance, but am unable to find it populated for ETFs. I forked it and found that if I add fundProfile module in quote.py on the _fetch_info call that it will populate `info['feesExpensesInvestment']['annualReportExpenseRatio']`.

Am I missing some functionality that will already allow me to access expense ratio for ETFs, or is this the best approach?

The method I updated in my fork:
`def _fetch_info(self, proxy):
        if self._already_fetched:
            return
        self._already_fetched = True
        modules = ['financialData', 'quoteType', 'defaultKeyStatistics', 'assetProfile', 'summaryDetail', 'fundProfile']`

**Comment by ValueRaider** [comment]
You're not missing anything.

Only criticism of your method is it will fetch `fundProfile` for all tickers not just ETFs, but being smarter requires a lot of work (adding quoteType to cache).

**Comment by yunhongz** [comment]
I reported an issue for it today. Just realize that it was discussed previously. Any progress?

**Reply by ValueRaider** [reply]
They wrote the code but never submitted a PR https://github.com/ntigern/yfinance/network
[How to contribute](#1084)

**Comment by ericpien** [comment]
Hi @ntigern, following https://github.com/ranaroussi/yfinance/pull/2041, you are now able to query `annualReportExpenseRatio` via:

```
spy = yf.Ticker('SPY')
data = spy.funds_data
data.fund_operations
```

Attributes | SPY | Category Average
-- | -- | --
Annual Report Expense Ratio | 0.000945 | 0.007769
Annual Holdings Turnover | 0.020000 | 0.490600
Total Net Assets | 308649.800000 | 308649.800000
